src/logic/sanitize.ts

```diff
@@ -24,3 +24,20 @@ export const sanitize = (text: string) => {
   if (!safeSummary) throw new Error(`No content after sanitizing`);
   return safeSummary;
 };
+
+//TODO: move this to a more appropriate place
+export const validateUrl = async (url: string) => {
+  
+  try {
+    // Check if the URL is valid
+    new URL(url);
+    
+    // Check if the api given URL can be reached
+    const response = await fetch(url, { method: 'HEAD' }); 
+    return response.ok;
+  } catch {
+    console.log('Invalid URL:',url);
```

Comment:
Probably makes sense for the caller to emit the log so it can add context like who the contact photo was for